src/werkzeug/formparser.py

```diff
@@ -461,6 +468,176 @@ def validate_boundary(self, boundary):
             # the assert is skipped.
             self.fail("Boundary longer than buffer size")
 
+    class LineSplitter:
+        def __init__(self, cap=None):
+            self.cap = cap
+            self._leftover_buffer = b""  # Holds lines to be used by next chunk of data
+
+        def _split_lines(self, input_buffer):
+            lines = input_buffer.splitlines(True)
+            cap_buffer = b""
+            for line in lines:
+                if self.cap:
+                    line = cap_buffer + line
+                    for low_bound in range(0, len(line), self.cap):
+                        capped_line = line[low_bound : low_bound + self.cap]
+                        if len(capped_line) >= self.cap:
+                            yield capped_line
+                    remainder = len(line) % self.cap
+                    cap_buffer = line[-remainder:]
+                elif line[-1:] in b"\r\n":
+                    yield line
+                else:
+                    self._leftover_buffer = line
+                    yield b""
+
+        def feed(self, input_buffer):
+            if not input_buffer:
+                yield self._leftover_buffer
+            else:
+                input_buffer = self._leftover_buffer + input_buffer
+
+                for line in self._split_lines(input_buffer):  # noqa: B007
+                    if line:
+                        yield line
+                        input_buffer = input_buffer[len(line) :]
```

Comment:
This doesn't do anything. `input_buffer` will go out of scope when this loop ends, so there's no reason to reassign it. What do you intend for this to do?